pandas/core/ops/common.py

```diff
@@ -122,3 +122,30 @@ def _maybe_match_name(a, b):
     elif b_has:
         return b.name
     return None
+
+
+def maybe_match_names_multiindex(a, b):
+    """
+    Try to find common names to attach to the result of an operation between
+    a and b.  Return a consensus list of names if they match at least partly
+    or None if they have completely different names.
```

Comment:
this could just be in padnas/core/indexes/multi.py i think would be fine

Reply:
Done